source/common/ur_util.hpp

```diff
@@ -13,6 +13,28 @@
 #include <string.h>
 #include <string>
 
+/* for compatibility with non-clang compilers */
+#if defined(__has_feature)
+#define CLANG_HAS_FEATURE(x) __has_feature(x)
+#else
+#define CLANG_HAS_FEATURE(x) 0
+#endif
+
+/* define for running with address sanitizer */
+#if CLANG_HAS_FEATURE(address_sanitizer) || defined(__SANITIZE_ADDRESS__)
```

Comment:
GCC also supports sanitizers, its not a clang only feature which this seems to imply.

Reply:
hmm, `HAS_FEATURE_IF_CLANG`? What I wanted to convey is that this macro checks for features if running under cmake - which I think is better than just having something generic like `HAS_FEATURE` or simply defining clang's `__has_feature__` if it's not present.